### PR TITLE
fix listing of current images

### PR DESCRIPTION
**Pull request by ieguinoa**
I'm deploying an instance to use docker as much as possible and ran into some problems while submitting jobs. These are the fixes I made, it works for me but not sure how correct it is.
Besides, sudo is set to False when building the command but it could be needed to retrieve the images list, shouldn't this be obtained from the job_conf.xml ?

### CODE REVIEW

File: lib/galaxy/tools/deps/container_resolvers/mulled.py [hunk 1]
```diff
@@ -54,8 +54,8 @@ def _package_hash(target):
 
 def list_docker_cached_mulled_images(namespace=None, hash_func="v2"):
     command = build_docker_images_command(truncate=True, sudo=False)
-    images_and_versions = subprocess.check_output(command).strip().split('\n')
-    images_and_versions = [line.split()[0:2] for line in images_and_versions[1:]]
+    images_and_versions = subprocess.check_output(command,shell=True).strip().split('\n')
```
Comment (bgruening) [inline lib/galaxy/tools/deps/container_resolvers/mulled.py:57]: @ieguinoa we should never use shell=True if possible, which limitation do you try to solve here?

Comment (ieguinoa) [reply lib/galaxy/tools/deps/container_resolvers/mulled.py:57]: It's just that the subprocess throws **OSError: [Errno 2] No such file or directory** if called with several parameters in a string and not a list. Shouldn't it be kept as a list when built and not parsed to string [here](https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/tools/deps/docker_util.py#L196)   ?

**Comment by martenson** [comment]
At the committers meeting we were unsure about the approach here and reluctant to use `shell=True` thus we are pushing this back to 18.05 to get more time for finding a better solution.

**Comment by jmchilton** [comment]
I missed the meeting - but this decision is frustrating. I'll try to find some time to fix it without shell=True and backport it - sorry @ieguinoa and thanks for trying to fix it.

**Comment by martenson** [comment]
@jmchilton You can also try to defend the choice, I think there was no sponsor of this at the meeting.

**Comment by jmchilton** [comment]
Thanks for the fix @ieguinoa - I've opened a back port of a fix for this issue to 17.09 with https://github.com/galaxyproject/galaxy/pull/5735. Hopefully that is merged quickly and I will push it to 18.01 and dev. Really sorry for missing this - I should looked more closely at this much sooner - thanks for at least trying to test out the mulled container options - let me know if you encounter any more problems or have any success 😓.

**Pull request closed without merging**